dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/navigation/NavResult.java

```diff
@@ -0,0 +1,374 @@
+            } catch (Exception e) {
+                titleToShow = title;
+            }
+            return "<a href='" + getHref() + "' title='" + titleToShow + "'>" + titleToShow + "</a>";
+        } else {
+            return getCodeLink();
+        }
+    }
+
+    public List<String> getChildrenFolderIds() {
+        return childrenFolderIds != null ? childrenFolderIds : new ArrayList<String>();
+    }
+
+    public void setChildrenFolderIds(List<String> childrenFolderIds) {
+        this.childrenFolderIds = childrenFolderIds;
+    }
+
+    @Override
+    public Iterator<NavResult> iterator() {
+        try {
+            List<? extends NavResult>  childs= getChildren();
+            List<NavResult> ret = new ArrayList<>();
+            for(NavResult child  : childs) {
+                ret.add(child);
+            }
+            return ret.iterator();
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public String getType() {
+        return type;
+    }
+
+    public void setType(String type) {
+        this.type = type;
+    }
+
+    public String getTarget() {
+        if (type.equals("link"))
+            return target;
+        else
+            return "_self";
+    }
+
+    public void setTarget(String target) {
+
+        this.target = target;
+    }
+
+    public String getPermissionId() {
+        return permissionId;
+    }
+
+    public void setPermissionId(String permissionId) {
+        this.permissionId = permissionId;
+    }
+
+
+    /// Permissionable methods ///
+
+    @Override
+    public String getOwner() {
+        return sysuser.getUserId();
+    }
+
+    @Override
+    public void setOwner(String owner) {}
+
+    @Override
+    public List<PermissionSummary> acceptedPermissions() {
+        return Arrays.asList(new PermissionSummary[] {new PermissionSummary("READ", "READ", PermissionAPI.PERMISSION_READ)});
+    }
+
+    @Override
+    public List<RelatedPermissionableGroup> permissionDependencies(int requiredPermission) {
+        return null;
+    }
+
+    @Override
+    public Permissionable getParentPermissionable() throws DotDataException {
+        try {
+            if (type.equals("htmlpage")) {
+                Identifier ident = APILocator.getIdentifierAPI()
+                    .find(permissionId);
+                if (Identifier.ASSET_TYPE_CONTENTLET.equalsIgnoreCase(ident.getAssetType()))
+                    return APILocator.getContentletAPI()
+                        .findContentletByIdentifier(permissionId, true, APILocator.getLanguageAPI()
+                            .getDefaultLanguage()
+                            .getId(), sysuser, false)
+                        .getParentPermissionable();
+            }
+            if (type.equals("folder"))
+                return APILocator.getFolderAPI()
+                    .find(folderId, sysuser, false)
+                    .getParentPermissionable();
+            if (type.equals("link"))
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Move the "link" string literal on the left side of this string comparison. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1132)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Missing curly brace. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00121)